### PR TITLE
Add test to check PARALLEL VACUUM functionality with PG13

### PR DESCRIPTION
**Pull request by nikkhils**
PG13 has introduced the parallel vacuum functionality in which if the
table has two or more indexes (with sizes beyond 512KB and enough number
of parallel worker processes), then the vacuum on those indexes can be
done in parallel.

### CODE REVIEW

File: test/expected/vacuum_parallel-13.out [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+CREATE INDEX vt_temp2 on vacuum_test (temp2);
+-- parallel vacuum needs the index size to be larger than 512KB to kick in
+INSERT INTO vacuum_test SELECT TIMESTAMP 'epoch' + (i * INTERVAL '100 second'),
+                i, i+1 FROM generate_series(1, 100000) as T(i);
+-- we should see two parallel workers for each chunk
+VACUUM (PARALLEL 3, VERBOSE) vacuum_test;
+INFO:  vacuuming "_timescaledb_internal._hyper_1_1_chunk"
+INFO:  launched 2 parallel vacuum workers for index cleanup (planned: 2)
+INFO:  index "_hyper_1_1_chunk_vacuum_test_time_idx" now contains 26279 row versions in 131 pages
+INFO:  index "_hyper_1_1_chunk_vt_temp1" now contains 26279 row versions in 74 pages
```
Comment (erimatnor) [inline test/expected/vacuum_parallel-13.out:27]: I am wondering if this output is fully predictable or if we need to mute it to ensure that the test won't fail under different circumstances (i.e., the  XX row versions in  YY pages would change depending on setting this is run under, leading to a test failure).

Comment (nikkhils) [reply test/expected/vacuum_parallel-13.out:27]: @erimatnor yeah, I thought of that. But I don't see a good way to mute the number of rows and pages info. Or is there something additional that I am not aware of that can help with this?

File: test/sql/.gitignore [hunk 2]
```diff
@@ -28,3 +28,4 @@
 /sql_query_results_x_diff-*.sql
 /sql_query-*.sql
 /update-*.sql
+/vacuum_parallel-*.sql
```
Comment (svenklemm) [inline test/sql/.gitignore:31]: this should not be part of this patch

**Review by svenklemm — COMMENTED** [review]
I suggest naming the test vacuum_parallel without version number since it will be needed for all versions >= 13

File: test/sql/.gitignore (hunk 2 above)
Comment (nikkhils) [reply test/sql/.gitignore:31]: yes, removed

**Comment by nikkhils** [comment]
> I suggest naming the test vacuum_parallel without version number since it will be needed for all versions >= 13

yeah, makes sense. Will rename.

**Review by pmwkaa — APPROVED** [review]
Looks good

File: test/expected/vacuum_parallel.out [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+CREATE INDEX vt_temp1 on vacuum_test (temp1);
+CREATE INDEX vt_temp2 on vacuum_test (temp2);
+-- parallel vacuum needs the index size to be larger than 512KB to kick in
+INSERT INTO vacuum_test SELECT TIMESTAMP 'epoch' + (i * INTERVAL '100 second'),
+                i, i+1 FROM generate_series(1, 100000) as T(i);
+-- we should see two parallel workers for each chunk
+VACUUM (PARALLEL 3, VERBOSE) vacuum_test;
+INFO:  vacuuming "_timescaledb_internal._hyper_1_1_chunk"
+INFO:  launched 2 parallel vacuum workers for index cleanup (planned: 2)
+INFO:  index "_hyper_1_1_chunk_vacuum_test_time_idx" now contains 26279 row versions in 131 pages
```
[the remark this answers is not in the document]
Comment (nikkhils) [reply test/expected/vacuum_parallel.out:26]: yeah, there are other tests which have similar flaky output and it seems to be holding up.

File: test/sql/vacuum_parallel.sql [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+-- create additional indexes on the temp columns
+CREATE INDEX vt_temp1 on vacuum_test (temp1);
+CREATE INDEX vt_temp2 on vacuum_test (temp2);
+
+-- parallel vacuum needs the index size to be larger than 512KB to kick in
+INSERT INTO vacuum_test SELECT TIMESTAMP 'epoch' + (i * INTERVAL '100 second'),
+                i, i+1 FROM generate_series(1, 100000) as T(i);
+
+-- we should see two parallel workers for each chunk
+VACUUM (PARALLEL 3, VERBOSE) vacuum_test;
```
[the remark this answers is not in the document]
Comment (nikkhils) [reply test/sql/vacuum_parallel.sql:24]: nothing multinode specific AFAICS here.

**Review by erimatnor — APPROVED** [review]
